FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -206,47 +228,138 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
         {
-            TimeSpan interval = TimeSpan.FromSeconds(IncrementalOrchestrationIntervalInSeconds);
+            OrchestratorJobStatusEntity jobStatus = await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, cancellationToken);
 
-            if (_inputData.DataStartTime == null || ((DateTimeOffset)_inputData.DataStartTime).AddMinutes(60) < _inputData.DataEndTime)
+            if (jobStatus == null)
             {
-                interval = TimeSpan.FromSeconds(InitialOrchestrationIntervalInSeconds);
+                var newJobStatus = new OrchestratorJobStatusEntity()
+                {
+                    PartitionKey = TableKeyProvider.JobStatusPartitionKey(_jobInfo.QueueType, Convert.ToInt32(JobType.Orchestrator)),
+                    RowKey = TableKeyProvider.JobStatusRowKey(_jobInfo.QueueType, Convert.ToInt32(JobType.Orchestrator), _jobInfo.GroupId),
+                    GroupId = _jobInfo.GroupId,
+                    StatisticResult = JsonConvert.SerializeObject(new FhirToDataLakeOrchestratorJobResult()),
+                };
+
+                if (!await _metadataStore.TryAddEntityAsync(newJobStatus, cancellationToken))
+                {
+                    throw new SynapsePipelineInternalException("Initialize job status failed");
+                }
+
+                jobStatus = await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, cancellationToken);
             }
 
-            while (_result.NextJobTimestamp == null || _result.NextJobTimestamp < _inputData.DataEndTime)
+            return jobStatus;
+        }
+
+        private async Task<OrchestratorJobStatusEntity> UpdateJobStatusWithResultAsync(CancellationToken cancellationToken)
+        {
+            _jobStatus.StatisticResult = JsonConvert.SerializeObject(_result);
+            if (!await _metadataStore.TryUpdateEntityAsync(_jobStatus))
             {
-                DateTimeOffset? lastEndTime = _result.NextJobTimestamp ?? _inputData.DataStartTime;
-                DateTimeOffset? nextResourceTimestamp =
-                    await GetNextTimestamp(lastEndTime, _inputData.DataEndTime, cancellationToken);
-                if (nextResourceTimestamp == null)
+                throw new SynapsePipelineInternalException("Update orchestrator job status failed.");
+            }
+
+            return await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, cancellationToken);
+        }
+
+        private async IAsyncEnumerable<FhirToDataLakeProcessingJobInputData> GetInputsAsyncForSystem([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // Resume the submitting job.
+            if (_result.SubmittingProcessingJob != null)
+            {
+                yield return JsonConvert.DeserializeObject<FhirToDataLakeProcessingJobInputData>(_result.SubmittingProcessingJob.First());
```

Comment:
Do we plan to support multiple SubmittingProcessingJob in the future.？

Reply:
Thanks! updated it and return a list here.